src/features/movies/movieSlice.ts

```diff
@@ -18,18 +18,26 @@ export const movieSlice = createSlice({
     initialState,
     reducers: {
         addMovie: (state, action: PayloadAction<IMovie>) => {
-            state.movies.push({                
-                text: action.payload.text,
-                rating: action.payload.rating,
-            })
+            let i = 0;
+            let ifDuplicate = false
+            while(i < state.movies.length){
+                if(state.movies[i].text === action.payload.text){
+                    ifDuplicate = true
+                    break;
+                }
+                i++
+            }
+            if (!ifDuplicate){
```

Comment:
How could we notify the user that the intended submission is a duplicate? Could we validate the input before dispatching the action?

Reply:
Added ui/ux for attempted submission. Also validated input before dispatching. Thanks for suggestions!